frontend/apps/reader/modules/readersearch.lua

```diff
@@ -51,33 +81,95 @@ function ReaderSearch:onShowFulltextSearchInput()
                     text = backward_text,
                     callback = function()
                         if self.input_dialog:getInputText() == "" then return end
-                        UIManager:close(self.input_dialog)
-                        self:onShowSearchDialog(self.input_dialog:getInputText(), 1)
+                        self.last_search_text = self.input_dialog:getInputText()
+                        self.use_regex = self.check_button_regex.checked
+                        self.case_insensitive = not self.check_button_case.checked
+                        if self.use_regex and self.ui.document:checkRegex(self.input_dialog:getInputText()) ~= 0 then
+                            UIManager:show(InfoMessage:new{ text = _("Invalid regular expression") })
+                        else
+                            UIManager:close(self.input_dialog)
+                            self:onShowSearchDialog(self.input_dialog:getInputText(), 1, self.use_regex, self.case_insensitive)
+                        end
                     end,
                 },
                 {
                     text = forward_text,
                     is_enter_default = true,
                     callback = function()
                         if self.input_dialog:getInputText() == "" then return end
-                        UIManager:close(self.input_dialog)
-                        self:onShowSearchDialog(self.input_dialog:getInputText(), 0)
+                        self.last_search_text = self.input_dialog:getInputText()
+                        self.use_regex = self.check_button_regex.checked
+                        self.case_insensitive = not self.check_button_case.checked
+                        if self.use_regex and self.ui.document:checkRegex(self.input_dialog:getInputText()) ~= 0 then
+                            UIManager:show(InfoMessage:new{ text = _("Invalid regular expression") })
```

Comment:
(Incidentally, there's no hint given by the library?)
```suggestion
                            UIManager:show(InfoMessage:new{ text = _("Invalid regular expression.") })
```

Reply:
There would be a lot of hints: error_escape, error_brack, error_range, error_badrepeat, error_brace, error_badbrace, error_paren, error_backref. 
Shall we all give an individual error message?

Reply:
Well, this is not the primary feature of the app, so I don't mind if we don't tell much (and go get all these warning messages translated !)
But you somehow made the checkRegex() function returns an integer and not a boolean :), so that's encourageing to get the backend return more info if it knows it (even if we don't yet show all the possible return code in the frontend).

Reply:
So this is the first change to crengine requested from the frontend :)
I will do that tomorrow morning. (return an int value ).